crates/router/tests/connectors/paypal.rs

```diff
@@ -0,0 +1,718 @@
+        .void_payment(
+            txn_id.unwrap(),
+            Some(types::PaymentsCancelData {
+                connector_transaction_id: String::from(""),
+                cancellation_reason: Some("requested_by_customer".to_string()),
+                connector_meta,
+                ..Default::default()
+            }),
+            get_default_payment_info(),
+        )
+        .await
+        .expect("Void payment response");
+    assert_eq!(void_response.response.unwrap_err().message, "description - Specified resource ID does not exist. Please check the resource ID and try again. ; ");
+}
+
+// Captures a payment using invalid connector payment id.
+#[actix_web::test]
+async fn should_fail_capture_for_invalid_payment() {
+    let connector_meta = Some(serde_json::json!({
+        "txn_id": "56YH8TZ",
+    }));
+    let capture_response = CONNECTOR
+        .capture_payment(
+            "123456789".to_string(),
+            Some(types::PaymentsCaptureData {
+                connector_meta,
+                ..utils::PaymentCaptureType::default().0
+            }),
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        capture_response.response.unwrap_err().message,
+        "description - Specified resource ID does not exist. Please check the resource ID and try again. ; ",
+    );
+}
+
+// Refunds a payment with refund amount higher than payment amount.
+#[actix_web::test]
+async fn should_fail_for_refund_amount_higher_than_payment_amount() {
+    let authorize_response = CONNECTOR
+        .authorize_payment(get_payment_data(), get_default_payment_info())
+        .await
+        .expect("Authorize payment response");
+    let txn_id = utils::get_connector_transaction_id(authorize_response.response.clone()).unwrap();
+    let capture_connector_meta = utils::get_connector_metadata(authorize_response.response);
+    let capture_response = CONNECTOR
+        .capture_payment(
+            txn_id.clone(),
+            Some(types::PaymentsCaptureData {
+                connector_meta: capture_connector_meta,
+                ..utils::PaymentCaptureType::default().0
+            }),
+            get_default_payment_info(),
+        )
+        .await
+        .expect("Capture payment response");
+    let refund_connector_metadata = utils::get_connector_metadata(capture_response.response);
+    let response = CONNECTOR
+        .refund_payment(
+            txn_id,
+            Some(types::RefundsData {
+                connector_metadata: refund_connector_metadata,
+                refund_amount: 150,
+                ..utils::PaymentRefundType::default().0
+            }),
+            get_default_payment_info(),
+        )
+        .await
+        .unwrap();
+    let re = Regex::new(r#"[{"description":"Specified resource ID does not exist. Please check the resource ID and try again.","field":"capture_id","issue":"INVALID_RESOURCE_ID","location":"path","value":".*"}]"#).unwrap();
```

Comment:
Why we are getting `INVALID_RESOURCE_ID ` in case of refund amount higher than the payment amount? Is this expected?